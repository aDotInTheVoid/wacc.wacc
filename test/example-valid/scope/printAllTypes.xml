<unit>
  <main>
    <addr_local>0</addr_local>
    <str>", "</str>
    <assign_do/>
    <addr_local>1</addr_local>
    <num>5</num>
    <assign_do/>
    <block>
      <addr_local>2</addr_local>
      <char>'x'</char>
      <assign_do/>
      <block>
        <addr_local>3</addr_local>
        <true/>
        <assign_do/>
        <block>
          <addr_local>4</addr_local>
          <str>"this is a string"</str>
          <assign_do/>
          <block>
            <addr_local>5</addr_local>
            <num>1</num>
            <num>2</num>
            <num>3</num>
            <array_lit>3</array_lit>
            <assign_do/>
            <block>
              <addr_local>6</addr_local>
              <char>'x'</char>
              <char>'y'</char>
              <char>'z'</char>
              <array_lit>3</array_lit>
              <assign_do/>
              <block>
                <addr_local>7</addr_local>
                <true/>
                <false/>
                <true/>
                <array_lit>3</array_lit>
                <assign_do/>
                <block>
                  <addr_local>8</addr_local>
                  <str>"array"</str>
                  <str>"of"</str>
                  <str>"strings"</str>
                  <array_lit>3</array_lit>
                  <assign_do/>
                  <block>
                    <addr_local>9</addr_local>
                    <num>1</num>
                    <num>2</num>
                    <newpair/>
                    <assign_do/>
                    <block>
                      <addr_local>10</addr_local>
                      <char>'a'</char>
                      <true/>
                      <newpair/>
                      <assign_do/>
                      <addr_local>11</addr_local>
                      <char>'b'</char>
                      <false/>
                      <newpair/>
                      <assign_do/>
                      <addr_local>12</addr_local>
                      <local>10</local>
                      <local>11</local>
                      <array_lit>2</array_lit>
                      <assign_do/>
                      <block>
                        <addr_local>13</addr_local>
                        <num>1</num>
                        <num>2</num>
                        <num>3</num>
                        <array_lit>3</array_lit>
                        <assign_do/>
                        <addr_local>14</addr_local>
                        <char>'a'</char>
                        <char>'b'</char>
                        <char>'c'</char>
                        <array_lit>3</array_lit>
                        <assign_do/>
                        <addr_local>15</addr_local>
                        <local>13</local>
                        <local>14</local>
                        <newpair/>
                        <assign_do/>
                        <block/>
                        <addr_local>16</addr_local>
                        <local>15</local>
                        <fst/>
                        <assign_do/>
                        <addr_local>17</addr_local>
                        <local>15</local>
                        <snd/>
                        <assign_do/>
                        <str>"( ["</str>
                        <print><kind>str</kind><newline>false</newline></print>
                        <local>16</local>
                        <num>0</num>
                        <array_elem/>
                        <print><kind>i32</kind><newline>false</newline></print>
                        <local>0</local>
                        <print><kind>str</kind><newline>false</newline></print>
                        <local>16</local>
                        <num>1</num>
                        <array_elem/>
                        <print><kind>i32</kind><newline>false</newline></print>
                        <local>0</local>
                        <print><kind>str</kind><newline>false</newline></print>
                        <local>16</local>
                        <num>2</num>
                        <array_elem/>
                        <print><kind>i32</kind><newline>false</newline></print>
                        <str>"] , ["</str>
                        <print><kind>str</kind><newline>false</newline></print>
                        <local>17</local>
                        <num>0</num>
                        <array_elem/>
                        <print><kind>char</kind><newline>false</newline></print>
                        <local>0</local>
                        <print><kind>str</kind><newline>false</newline></print>
                        <local>17</local>
                        <num>1</num>
                        <array_elem/>
                        <print><kind>char</kind><newline>false</newline></print>
                        <local>0</local>
                        <print><kind>str</kind><newline>false</newline></print>
                        <local>17</local>
                        <num>2</num>
                        <array_elem/>
                        <print><kind>char</kind><newline>false</newline></print>
                        <str>"] )"</str>
                        <print><kind>str</kind><newline>true</newline></print>
                      </block>
                      <addr_local>13</addr_local>
                      <local>12</local>
                      <num>0</num>
                      <array_elem/>
                      <assign_do/>
                      <addr_local>14</addr_local>
                      <local>13</local>
                      <fst/>
                      <assign_do/>
                      <addr_local>15</addr_local>
                      <local>13</local>
                      <snd/>
                      <assign_do/>
                      <addr_local>16</addr_local>
                      <local>12</local>
                      <num>1</num>
                      <array_elem/>
                      <assign_do/>
                      <addr_local>17</addr_local>
                      <local>16</local>
                      <fst/>
                      <assign_do/>
                      <addr_local>18</addr_local>
                      <local>16</local>
                      <snd/>
                      <assign_do/>
                      <str>"[ "</str>
                      <print><kind>str</kind><newline>false</newline></print>
                      <local>13</local>
                      <print><kind>ptr</kind><newline>false</newline></print>
                      <str>" = ("</str>
                      <print><kind>str</kind><newline>false</newline></print>
                      <local>14</local>
                      <print><kind>char</kind><newline>false</newline></print>
                      <local>0</local>
                      <print><kind>str</kind><newline>false</newline></print>
                      <local>15</local>
                      <print><kind>bool</kind><newline>false</newline></print>
                      <str>"), "</str>
                      <print><kind>str</kind><newline>false</newline></print>
                      <local>16</local>
                      <print><kind>ptr</kind><newline>false</newline></print>
                      <str>" = ("</str>
                      <print><kind>str</kind><newline>false</newline></print>
                      <local>17</local>
                      <print><kind>char</kind><newline>false</newline></print>
                      <local>0</local>
                      <print><kind>str</kind><newline>false</newline></print>
                      <local>18</local>
                      <print><kind>bool</kind><newline>false</newline></print>
                      <str>") ]"</str>
                      <print><kind>str</kind><newline>true</newline></print>
                    </block>
                    <addr_local>10</addr_local>
                    <local>9</local>
                    <fst/>
                    <assign_do/>
                    <addr_local>11</addr_local>
                    <local>9</local>
                    <snd/>
                    <assign_do/>
                    <local>10</local>
                    <print><kind>i32</kind><newline>false</newline></print>
                    <local>0</local>
                    <print><kind>str</kind><newline>false</newline></print>
                    <local>11</local>
                    <print><kind>i32</kind><newline>true</newline></print>
                  </block>
                  <addr_local>9</addr_local>
                  <local>8</local>
                  <num>0</num>
                  <array_elem/>
                  <assign_do/>
                  <addr_local>10</addr_local>
                  <local>8</local>
                  <num>1</num>
                  <array_elem/>
                  <assign_do/>
                  <addr_local>11</addr_local>
                  <local>8</local>
                  <num>2</num>
                  <array_elem/>
                  <assign_do/>
                  <local>9</local>
                  <print><kind>str</kind><newline>false</newline></print>
                  <local>0</local>
                  <print><kind>str</kind><newline>false</newline></print>
                  <local>10</local>
                  <print><kind>str</kind><newline>false</newline></print>
                  <local>0</local>
                  <print><kind>str</kind><newline>false</newline></print>
                  <local>11</local>
                  <print><kind>str</kind><newline>true</newline></print>
                </block>
                <local>7</local>
                <num>0</num>
                <array_elem/>
                <print><kind>bool</kind><newline>false</newline></print>
                <local>0</local>
                <print><kind>str</kind><newline>false</newline></print>
                <local>7</local>
                <num>1</num>
                <array_elem/>
                <print><kind>bool</kind><newline>false</newline></print>
                <local>0</local>
                <print><kind>str</kind><newline>false</newline></print>
                <local>7</local>
                <num>2</num>
                <array_elem/>
                <print><kind>bool</kind><newline>true</newline></print>
              </block>
              <local>6</local>
              <print><kind>char_array</kind><newline>true</newline></print>
            </block>
            <addr_local>6</addr_local>
            <local>5</local>
            <num>0</num>
            <array_elem/>
            <assign_do/>
            <addr_local>7</addr_local>
            <local>5</local>
            <num>1</num>
            <array_elem/>
            <assign_do/>
            <addr_local>8</addr_local>
            <local>5</local>
            <num>2</num>
            <array_elem/>
            <assign_do/>
            <local>6</local>
            <print><kind>i32</kind><newline>false</newline></print>
            <local>0</local>
            <print><kind>str</kind><newline>false</newline></print>
            <local>7</local>
            <print><kind>i32</kind><newline>false</newline></print>
            <local>0</local>
            <print><kind>str</kind><newline>false</newline></print>
            <local>8</local>
            <print><kind>i32</kind><newline>true</newline></print>
          </block>
          <local>4</local>
          <print><kind>str</kind><newline>true</newline></print>
        </block>
        <local>3</local>
        <print><kind>bool</kind><newline>true</newline></print>
      </block>
      <local>2</local>
      <print><kind>char</kind><newline>true</newline></print>
    </block>
    <local>1</local>
    <print><kind>i32</kind><newline>true</newline></print>
  </main>
</unit>