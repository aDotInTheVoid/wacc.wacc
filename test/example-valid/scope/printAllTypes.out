( [1, 2, 3] , [a, b, c] )
[ 0x{{PTR}} = (a, true), 0x{{PTR}} = (b, false) ]
1, 2
array, of, strings
true, false, true
xyz
1, 2, 3
this is a string
true
x
5
